rewriter/Initializer.cc

```diff
@@ -140,46 +134,13 @@ void checkSigReturnType(core::MutableContext ctx, const ast::Send *send) {
         return;
     }
 
-    if (auto e = ctx.beginError(originalSendLoc, core::errors::Rewriter::InitializeReturnType)) {
+    auto errLoc = (send->funLoc.exists() && !send->funLoc.empty()) ? send->funLoc : send->loc;
+    if (auto e = ctx.beginError(errLoc, core::errors::Rewriter::InitializeReturnType)) {
         e.setHeader("The {} method should always return {}", "initialize", "void");
-
-        auto loc = core::Loc(ctx.file, originalSendLoc);
-        auto original = string(loc.source(ctx).value());
-        unsigned long returnsStart = original.find("returns");
-        unsigned long returnsLength, afterReturnsPosition;
-        string replacement;
-        string statementAfterReturns = "";
-        if (funAfterReturns.exists()) {
-            statementAfterReturns = funAfterReturns.toString(ctx);
-        }
-
-        // If there are no statements after returns(), we can use the length of the block to find the length
-        // we need to replace. If there are statements after it, we need to find the exact length using the next
-        // statement and remember to add a dot or else it will produce invalid code
-        returnsLength = original.length() - returnsStart + 1;
-
-        if (statementAfterReturns.empty()) {
-            replacement = original.replace(returnsStart, returnsLength, "void");
-        } else {
-            afterReturnsPosition = original.find(statementAfterReturns, returnsStart);
-
-            // If there is a line break between returns() and the next statement, change the returns() entry and
-            // re-join the string with the line breaks. Otherwise, everything is on the same line and we can replace
-            // directly without worrying about line breaks
-            vector<string> lines = absl::StrSplit(original.substr(returnsStart, afterReturnsPosition), "\n");
-
-            if (lines.size() > 1) {
-                lines[0] = "void";
-                replacement = original.replace(returnsStart, returnsLength,
-                                               fmt::format("{}", fmt::join(lines.begin(), lines.end(), "\n")));
-            } else {
-                returnsLength = original.find(statementAfterReturns, returnsStart) - returnsStart;
-                replacement = original.replace(returnsStart, returnsLength, "void.");
-            }
+        auto replacementLoc = core::Loc(ctx.file, send->funLoc.beginPos(), send->loc.endPos());
+        if (replacementLoc.exists() && !replacementLoc.empty()) {
+            e.replaceWith("Replace `.returns` with `.void`", replacementLoc, "void");
```

Comment:
The original change was contributed by an external contributor which explains why the code was going through things in such a roundabout and brittle way.